OneOfDiagnosticSuppressor/ExpressionHelper.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis;
+
+namespace SvSoft.OneOf.Analyzers.SwitchDiagnosticSuppression;
+
+static class ExpressionHelper
+{
+    static ExpressionSyntax GetUnparenthesizedExpression(ExpressionSyntax expression)
+    {
+        while (expression is ParenthesizedExpressionSyntax parenthesizedExpression)
+            expression = parenthesizedExpression.Expression;
+
+        return expression;
+    }
+
+    static ITypeSymbol? GetTypeSymbolFromIdentifier(SemanticModel model, IdentifierNameSyntax syntax)
+        => model.GetTypeInfo(syntax).Type;
+
+    static ITypeSymbol? GetTypeSymbolFromInvocation(SemanticModel model, InvocationExpressionSyntax invocation, bool expectTaskReturnType)
+    {
+        var invokedSymbol = model.GetSymbolInfo(invocation);
+        if (invokedSymbol.Symbol is not IMethodSymbol methodSymbol || methodSymbol.ReturnType is not INamedTypeSymbol returnTypeSymbol)
```

Comment:
Could be simplified to `invokedSymbol.Symbol is not IMethodSymbol { ReturnType: INamedTypeSymbol returnTypeSymbol }`